src/schematools/events/partial.py

```diff
@@ -0,0 +1,549 @@
+        if modification["key"] == "geometrie":
+            continue
+        update_data[modification["key"]] = modification["new_value"]
+    return update_data
+
+
+FK_TABLE = False
+NM_TABLE = True  # nm_table_id is not None
+
+
+# Configuration information to map the event-type
+# to the following fields (package into a DbInfo dataclass):
+# - db_operation_name (INSERT/UPDATE)
+# - needs_select: generated SQL needs to select a record
+# - needs_values: generated SQL needs values (to insert/update)
+# - data_fetcher: function to be used to extract data from the event
+
+EVENT_TYPE_MAPPINGS = {
+    FK_TABLE: {
+        "ADD": ("update", True, True, fetch_insert_data),
+        "MODIFY": ("update", True, True, fetch_update_data),
+        "DELETE": ("update", True, True, None),
+    },
+    NM_TABLE: {
+        "ADD": ("insert", False, True, fetch_insert_data),
+        "MODIFY": ("update", True, True, fetch_update_data),
+        "DELETE": ("delete", True, False, None),
+    },
+}
+
+
+@dataclass
+class DbInfo:
+    db_operation_name: str
+    needs_select: bool
+    needs_values: bool
+    data_fetcher: Optional[Callable[[Dict], Dict]]
+
+    @classmethod
+    def from_event_type(cls, is_nm_table: bool, event_type: str):
+        return cls(*EVENT_TYPE_MAPPINGS[is_nm_table][event_type])
+
+
+# Translation table of GOB relation names to amsterdam schema configuration
+# Events for relations contain a 'rel' field that is an internal GOB name
+# We need to associate configuration info with these events to be able
+# to map the event to the correct tables according to the amsterdam schema.
+
+# Structure
+# <gob rel name> :
+#       (<amschema dataset>, <amschema table>, <nm relation table>, <amschema fieldname>)
+
+COLLECTION_TO_SCHEMA = {
+    "gbd_bbk_gbd_brt_ligt_in_buurt": ("gebieden", "bouwblokken", None, "ligt_in_buurt"),
+    "gbd_brt_gbd_wijk_ligt_in_wijk": ("gebieden", "buurten", None, "ligt_in_wijk"),
+    "gbd_ggp_gbd_sdl_ligt_in_stadsdeel": (
+        "gebieden",
+        "ggpgebieden",
+        None,
+        "ligt_in_stadsdeel",
+    ),
+    "gbd_ggw_gbd_sdl_ligt_in_stadsdeel": (
+        "gebieden",
+        "ggwgebieden",
+        None,
+        "ligt_in_stadsdeel",
+    ),
+    "gbd_wijk_gbd_sdl_ligt_in_stadsdeel": (
+        "gebieden",
+        "wijken",
+        None,
+        "ligt_in_stadsdeel",
+    ),
+    "gbd_ggw_gbd_brt_bestaat_uit_buurten": (
+        "gebieden",
+        "ggwgebieden",
+        "gebieden_ggwgebieden_bestaat_uit_buurten",
+        "bestaat_uit_buurten",
+    ),
+    "gbd_ggp_gbd_brt_bestaat_uit_buurten": (
+        "gebieden",
+        "ggpgebieden",
+        "gebieden_ggpgebieden_bestaat_uit_buurten",
+        "bestaat_uit_buurten",
+    ),
+}
+
+# Special fields, unable to query
+# gbd_brt_gbd_ggp__ligt_in_ggpgebied
+# gbd_brt_gbd_ggw__ligt_in_ggwgebied
+# gbd_wijk_gbd_ggw__ligt_in_ggwgebied
+
+# Relatie buiten gebieden
+# gbd_sdl_brk_gme_ligt_in_gemeente
+
+
+@dataclass
+class SchemaInfo:
```

Comment:
Wat doet dit object? Wat is het doel? Waarvoor bestaat het? Dat is mij erg onduidelijk, en zou een hoop "cognitieve load" van de code verlagen.